source/handy_httpd/components/websocket/frame.d

```diff
@@ -223,11 +223,11 @@ private bool validateOpcode(ubyte opcode) {
  *   stream = The stream to read from.
  * Returns: The complete payload length.
  */
-private ulong readPayloadLength(S)(ubyte initialLength, S stream) if (isByteInputStream!S) {
+private size_t readPayloadLength(S)(ubyte initialLength, S stream) if (isByteInputStream!S) {
     if (initialLength == 126) {
         return readDataOrThrow!(ushort)(stream);
     } else if (initialLength == 127) {
-        return readDataOrThrow!(ulong)(stream);
+        return readDataOrThrow!(size_t)(stream);
```

Comment:
The specification requires that the payload length be encoded as 64 bits if the initial length is 127, so it's gotta stay as `readDataOrThrow!(ulong)(stream)`. I guess the best thing to do would be `return cast(size_t) readDataOrThrow!(ulong)(stream);`

Reply:
you are right, the frame length could be [64bit](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#decoding_payload_length),  but if I use ulong in 32 arm the compiler complains. I could do a cast from ulong to uint, but if actually the package is bigger than uint it generates a conversion error.... I have to think more about it. Thanks for the great support